### PR TITLE
Add filter pipe

### PR DESCRIPTION
**Pull request by blacksam07**
## Summary
- Add pipe to filter books

## Trello card
[23-pipes](https://trello.com/c/NpEr08PX/23-pipes)

## Screenshots
pipe-min

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { AuthModule } from './modules/auth/auth.module';
 import { UnauthGuard } from './guards/unauth.guard';
 import { UnauthModule } from './modules/unauth/unauth.module';
 import { TokenInterceptor } from './token.interceptor';
+import { TitleFilterPipe } from './pipes/title-filter.pipe';
```
Comment (CrisVega08) [inline src/app/app.module.ts]: create a module for pipes

File: src/app/pipes/title-filter.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'titleFilter'
```
Comment (CrisVega08) [inline src/app/pipes/title-filter.pipe.ts]:
```suggestion
  name: 'filter'
```

File: src/app/screens/book-list/book-list.component.html [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
-    <div class="list-item">
-      <a class="item-book-cover"></a>
-      <span class="item-bold-text">{{book.title}}</span>
-      <span class="item-text">{{book.author}}</span>
-    </div>
-  </ng-template>
+<div class="container">
+  <input [(ngModel)]="filter" #ctrl="ngModel" class="filter" type="text" name="filter" >
+  <div class="list-container">
+    <ng-template ngFor let-book [ngForOf]="books | titleFilter:filter" let-i="index">
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.html]: When did you learned this weird way of loop? change please and you dont need index, also delete ng-template

File: src/app/screens/book-list/book-list.component.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, AfterContentInit } from '@angular/core';
+import { Component, OnInit, AfterContentInit, OnChanges, SimpleChanges, DoCheck } from '@angular/core';
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.ts]: delete onchanges, simplechanges and docheck

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/screens/book-list/book-list.component.html (hunk 3 above)
Comment (blacksam07) [reply src/app/screens/book-list/book-list.component.html]: In official documentation https://angular.io/api/common/NgForOf

Comment (betoobernal) [reply src/app/screens/book-list/book-list.component.html]: 😆

File: src/app/pipes/title-filter.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+})
+export class TitleFilterPipe implements PipeTransform {
```
Comment (betoobernal) [inline src/app/pipes/title-filter.pipe.ts]: Change the name class, also.

File: src/app/modules/pipes/filter.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filter'
+})
+export class FilterPipe implements PipeTransform {
+
+  transform(items: any[], filterText: string): any[] {
+    if(filterText !== '') {
```
Comment (CrisVega08) [inline src/app/modules/pipes/filter.pipe.ts]:
```suggestion
    if(filterText) {
```

File: src/app/modules/pipes/filter.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      return items.filter(item => item.title.toLowerCase().match(filterText.toLowerCase()) !== null);
```
Comment (CrisVega08) [inline src/app/modules/pipes/filter.pipe.ts]: use ternary

File: src/app/modules/pipes/pipes.module.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+import { FilterPipe } from './filter.pipe';
+
+@NgModule({
+  declarations: [FilterPipe],
+  imports: [
+    CommonModule
+  ],
+  exports: [FilterPipe]
+})
+export class PipesModule { }
```
Comment (CrisVega08) [inline src/app/modules/pipes/pipes.module.ts]:
```suggestion
export class PipesModule {}
```

File: src/app/screens/book-list/book-list.component.html [hunk 9]
```diff
@@ -1,7 +1,10 @@
-<div class="list-container">
-  <div *ngFor="let book of books"  class="list-item">
-    <a class="item-book-cover"></a>
-    <span class="item-bold-text">{{book.title}}</span>
-    <span class="item-text">{{book.author}}</span>
+<div class="container">
+  <input [(ngModel)]="filterValue" #ctrl="ngModel" class="filter" type="text" name="filter" >
+  <div class="list-container">
+    <div *ngFor="let book of books | filter:filterValue"  class="list-item">
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.html]:
```suggestion
    <div *ngFor="let book of books | filter:filterValue" class="list-item">
```

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/screens/book-list/book-list.component.scss [hunk 10]
```diff
@@ -1,12 +1,28 @@
...
+}
+
+.filter {
+  box-sizing: border-box;
+  height: 50px;
+  width: 645px;
+  border-radius: 5px;
+  border: 2px solid $mercury;
+  background: url('~/assets/search.png') no-repeat scroll 98% 50%;
+  padding-right: 45px;
```
Comment (betoobernal) [inline src/app/screens/book-list/book-list.component.scss]: order

**Review by betoobernal — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**